app/controllers/participants_controller.rb

```diff
@@ -12,10 +12,19 @@ def assign_number
   end
 
   def create
-    participant = Participant.create! user_params
-    redirect_to participants_path,
-                notice: 'Participant has been created.'
-    PrivatePub.publish_to '/scoreboard', participant
+    participant = Participant.new user_params
+    if participant.save
+      redirect_to participants_path,
+        notice: 'Participant has been created.'
+      PrivatePub.publish_to '/scoreboard', participant
+    elsif participant.errors.include?(:name)
+      redirect_to participants_path,
+        alert: 'Participant with that name already exists!'
+    # add more failures here if necessary
```

Comment:
It'd be good to do it more dynamically than this.

[Here](https://github.com/umts/jobapps/blob/master/app/controllers/application_controller.rb#L63) is an [example](https://github.com/umts/jobapps/blob/master/app/views/layouts/application.haml#L22-L27).